axelor-business-project/src/main/resources/views/ProjectTask.xml

```diff
@@ -66,4 +95,39 @@
       expr="eval: __config__.app.getNbDecimalDigitForUnitPrice()"/>
   </action-attrs>
 
+  <action-group name="action-business-project-project-task-group-on-new">
+    <action name="action-project-task-group-onnew"/>
+    <action name="action-project-task-attrs-set-time-unit-default"/>
+    <action name="action-project-task-attrs-set-time-unit-domain"/>
+  </action-group>
+
+  <action-group name="action-business-project-project-task-group-on-load">
+    <action name="action-project-task-group-timer-process"/>
+    <action name="action-project-task-attrs-set-time-unit-domain"/>
+  </action-group>
+
+  <!-- get timeUnit value from configuration -->
+  <action-attrs name="action-project-task-attrs-set-time-unit-default">
+    <attribute for="timeUnit" name="value"
+      expr="eval: __config__.app.getApp('business-project')?.daysUnit"/>
+  </action-attrs>
+
+  <!-- set timeUnit domain onNew and onLoad -->
+  <action-attrs name="action-project-task-attrs-set-time-unit-domain">
+    <attribute for="timeUnit" name="domain"
+      expr="eval: &quot; self.id = ${__config__.app.getApp('business-project')?.daysUnit?.id} OR self.id = ${__config__.app.getApp('business-project')?.hoursUnit?.id}&quot;"/>
+  </action-attrs>
```

Comment:
please call this action `onSelect` instead of `onNew` and `onLoad` as this is the standard way to set a domain